frontend/src/pages/dashboard/CampaignGrid/index.js

```diff
@@ -20,7 +21,7 @@ const CampaignGrid = ({ campaigns }) => {
             startDate={dateToString(new Date(campaign.campaign.starts_at))}
             endDate={dateToString(new Date(campaign.campaign.ends_at))}
             img={bytesToImage(campaign.campaign.cover_image)}
-            applyClick={() => navigate("/application")}
+            applyClick={() => navigate(`/application`, { state: campaign })}
```

Comment:
It would probably be better to use a URL parameter here with the campaign id so it's actually part of the route and we can navigate directly to the link or refresh the page without it breaking. You'll have to change the route in `src/routes.js` for this. You can look at some of the others in here for examples.

Maybe you can still give the whole campaign in state like this for the navigation here (in addition to having campaign ID in url) and only manually fetch the campaign from ID if it doesn't also have the state (i.e., direct navigation/refresh)